M2/Macaulay2/m2/version.m2.in

```diff
@@ -1,9 +0,0 @@
-srcversion := "@PACKAGE_VERSION@"
-binversion := version#"VERSION"
-if srcversion =!= binversion
-then stderr << "--warning: mismatch: source code version: " << srcversion << " != executable version " << binversion << endl
```

Comment:
It has been _years_ since this file was generated, distributed, or loaded, and nothing has gone wrong, so I think we can live without it.

Reply:
Hmm, you're right.  I wonder why I disabled it in 2012.  In any case, I'd prefer to reinstate it, as it still seems important.

```diff
commit e44155cf1257344c004466372d11dc1b656abf19
Author: Daniel Grayson <dan@math.uiuc.edu>
Date:   Tue Mar 20 20:46:34 2012 +0000

    stop reading the manufactured file "version.m2"
```

Reply:
I'd rather not. It's useless and hinders development. I frequently have to use sources compiled with different but close versions of M2 (e.g. M2 1.17 using Core from Github or M2 built from source running on the distributed Core) and this would only cause pain.

We've gone 9 years without any issues. If we run into any issues you can add it in a future PR. For now I'm removing it as dead code.

Reply:
I'm marking this as resolved. If you'd like to add and enable it again, I think you should make a new issue so we can discuss why it is needed and how the version should be checked there.

Reply:
I want my code to stay.

Reply:
Okay, you can fix it and add it again. These 4 lines of dead code have been unreachable for ~9 years and you have yet to explain why they are so important and how you didn't notice that they weren't used for so long if they are so important.